packages/ember-glimmer/lib/syntax.js

```diff
@@ -1,44 +1,85 @@
-import { RenderSyntax } from './syntax/render';
-import { OutletSyntax } from './syntax/outlet';
-import { MountSyntax } from './syntax/mount';
-import { DynamicComponentSyntax } from './syntax/dynamic-component';
-import { InputSyntax } from './syntax/input';
+import { renderMacro } from './syntax/render';
+import { outletMacro } from './syntax/outlet';
+import { mountMacro } from './syntax/mount';
 import {
-  WithDynamicVarsSyntax,
-  InElementSyntax
-} from 'glimmer-runtime';
+  blockComponentMacro,
+  inlineComponentMacro,
+  closureComponentMacro
+} from './syntax/dynamic-component';
+import { wrapComponentClassAttribute } from './utils/bindings';
+import { _withDynamicVarsMacro } from './syntax/-with-dynamic-vars';
+import { _inElementMacro } from './syntax/-in-element';
+import { inputMacro } from './syntax/input';
+import { textAreaMacro } from './syntax/-text-area';
+import { assert } from 'ember-metal';
 
+function refineInlineSyntax(path, params, hash, builder) {
+  let [ name ] = path;
 
-let syntaxKeys = [];
-let syntaxes = [];
+  assert(`You attempted to overwrite the built-in helper "${name}" which is not allowed. Please rename the helper.`, !(builder.env.builtInHelpers[name] && builder.env.owner.hasRegistration(`helper:${name}`)));
 
-export function registerSyntax(key, syntax) {
```

Comment:
I would still like to export a mechanism to register macros. Without one we loose the ability to experiment with different syntaxes (e.g. `ember-let`). 

It seems totally good to change it to a more generic thing like `registerMacro`, but we shouldn't remove the ability to customize/experiment.